rest/data/aliases.go

```diff
@@ -131,6 +132,47 @@ func UpdateAliasesForSection(projectId string, updatedAliases []restModel.APIPro
 	return modified, catcher.Resolve()
 }
 
+// validateFeaturesHaveAliases returns an error if project/repo aliases are not defined for a Github/CQ feature.
+// Does not error if version control is enabled.
+func validateFeaturesHaveAliases(pRef *model.ProjectRef, aliases []restModel.APIProjectAlias) error {
+	if pRef.IsVersionControlEnabled() {
+		return nil
+	}
+
+	aliasesMap := map[string]bool{}
+	for _, a := range aliases {
+		aliasesMap[utility.FromStringPtr(a.Alias)] = true
+	}
+
+	if pRef.RepoRefId != "" {
+		repoAliases, err := model.FindAliasesForRepo(pRef.RepoRefId)
+		if err != nil {
+			return err
+		}
+		for _, a := range repoAliases {
+			aliasesMap[a.Alias] = true
+		}
+
+	}
+
+	msg := "%s cannot be enabled without aliases"
```

Comment:
'%s'? even though it'll never be an empty string

Reply:
I'd rather not actually, since this is purely for string formatting purposes and not wrapping an ID or something where it isn't clear where it starts.